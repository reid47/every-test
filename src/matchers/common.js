export const formatValue = received => {
  if (received == null) return String(received);
  if (received instanceof HTMLElement) return received.outerHTML;

  try {
    return JSON.stringify(received);
  } catch {
    return received.toString();
  }
};
