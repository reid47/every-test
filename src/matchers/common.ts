export const formatValue = (received: any): string => {
  if (received == null) return String(received);
  if (received instanceof HTMLElement) return received.outerHTML;

  try {
    return JSON.stringify(received);
  } catch {
    return received.toString();
  }
};
