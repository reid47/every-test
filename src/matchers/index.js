export { default as toBeChecked } from './toBeChecked';
export { default as toBeDisabled } from './toBeDisabled';
export { default as toBeFocused } from './toBeFocused';
export { default as toBeHidden } from './toBeHidden';
export { default as toContainText } from './toContainText';
export { default as toExist } from './toExist';
export { default as toHaveAttr } from './toHaveAttr';
export { default as toHaveClass } from './toHaveClass';
export { default as toHaveProp } from './toHaveProp';
export { default as toHaveText } from './toHaveText';
export { default as toHaveValue } from './toHaveValue';
