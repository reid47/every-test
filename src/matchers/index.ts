export { default as toBeChecked } from './toBeChecked';
export { default as toBeDisabled } from './toBeDisabled';
export { default as toBeEmpty } from './toBeEmpty';
export { default as toBeFocused } from './toBeFocused';
export { default as toBeHidden } from './toBeHidden';
export { default as toContainText } from './toContainText';
export { default as toExist } from './toExist';
export { default as toHaveAttribute } from './toHaveAttribute';
export { default as toHaveClass } from './toHaveClass';
export { default as toHaveTagName } from './toHaveTagName';
export { default as toHaveText } from './toHaveText';
export { default as toHaveValue } from './toHaveValue';
