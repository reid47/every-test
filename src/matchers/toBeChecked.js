export default function toBeChecked(received) {
  const pass = !!received.checked;
  const message = pass ? `Expected element not to be checked.` : `Expected element to be checked.`;
  return { pass, message };
}
