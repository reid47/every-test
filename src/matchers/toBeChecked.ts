export default function toBeChecked(received: HTMLElement) {
  const pass = received instanceof HTMLInputElement && received.checked;
  const message = pass ? `Expected element not to be checked.` : `Expected element to be checked.`;
  return { pass, message };
}
