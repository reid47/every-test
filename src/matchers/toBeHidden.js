export default function toBeHidden(received) {
  const hiddenProp = !!received.hidden;
  const { visibility, width, height } = getComputedStyle(received);
  const computedVisibility = visibility === 'hidden';
  const zeroWidth = width <= 0;
  const zeroHeight = height <= 0;

  const pass = hiddenProp || computedVisibility || zeroWidth || zeroHeight;

  const message = pass
    ? 'Expected element not to be hidden, but it was.'
    : 'Expected element to be hidden, but it was not.';

  return { pass, message };
}
