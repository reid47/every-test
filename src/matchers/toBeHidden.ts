export default function toBeHidden(received) {
  const hiddenProp = !!received.hidden;
  const { visibility, width, height } = getComputedStyle(received);
  const computedVisibility = visibility === 'hidden';
  const zeroWidth = width ? parseInt(width) <= 0 : true;
  const zeroHeight = height ? parseInt(height) <= 0 : true;

  const pass = hiddenProp || computedVisibility || zeroWidth || zeroHeight;

  const message = pass
    ? 'Expected element not to be hidden, but it was.'
    : 'Expected element to be hidden, but it was not.';

  return { pass, message };
}
