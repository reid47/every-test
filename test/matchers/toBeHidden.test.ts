import { toBeHidden } from '../../src/matchers/toBeHidden';

describe('toBeHidden', () => {
  test('passing (hidden prop)', () => {
    const el = document.createElement('div');
    const text = document.createTextNode('hello');
    el.appendChild(text);
    el.hidden = true;

    const result = toBeHidden(el);
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('passing (display: none)', () => {
    const el = document.createElement('div');
    const text = document.createTextNode('hello');
    el.appendChild(text);
    el.style.display = 'none';

    const result = toBeHidden(el);
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('passing (zero-width)', () => {
    const el = document.createElement('div');
    const text = document.createTextNode('hello');
    el.appendChild(text);
    el.style.width = '0px';

    const result = toBeHidden(el);
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('passing (zero-height)', () => {
    const el = document.createElement('div');
    const text = document.createTextNode('hello');
    el.appendChild(text);
    el.style.height = '0px';

    const result = toBeHidden(el);
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing', () => {
    const el = document.createElement('div');
    const text = document.createTextNode('hello');
    el.appendChild(text);
    el.style.width = '10px';
    el.style.height = '10px';

    const result = toBeHidden(el);
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();
  });
});
