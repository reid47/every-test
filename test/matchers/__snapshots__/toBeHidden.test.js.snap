// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeHidden failing 1`] = `"Expected element to be hidden, but it was not."`;

exports[`toBeHidden passing (display: none) 1`] = `"Expected element not to be hidden, but it was."`;

exports[`toBeHidden passing (hidden prop) 1`] = `"Expected element not to be hidden, but it was."`;

exports[`toBeHidden passing (zero-height) 1`] = `"Expected element not to be hidden, but it was."`;

exports[`toBeHidden passing (zero-width) 1`] = `"Expected element not to be hidden, but it was."`;
